lightning-background-processor/src/lib.rs

```diff
@@ -303,6 +303,18 @@ mod tests {
 		}}
 	}
 
+	fn connect_blocks(node: &mut Node, block_count: u32) {
+		for i in 1..=block_count {
+			let prev_blockhash = node.best_block.block_hash();
+			let height = node.best_block.height() + 1;
+			let header = BlockHeader { version: 0x20000000, prev_blockhash, merkle_root: Default::default(), time: height, bits: 42, nonce: 42 };
+			node.best_block = BestBlock::new(header.block_hash(), height);
+			if i == block_count {
+				node.node.best_block_updated(&header, height);
+				node.chain_monitor.best_block_updated(&header, height);
+			}
+		}
+	}
```

Comment:
Would it be worth de-duplicating `confirm_transaction` and `connect_blocks`?

Reply:
Right, I stared at it a bit but it looked like de-duplicating would have as much boilerplate as code it would avoid. Not sure if you had something specific in mind.

Reply:
I suppose you can instead just use `confirm_transaction` but replace `ANTI_REORG_DELAY` with `BREAKDOWN_TIMEOUT as u32 - 1`.

Reply:
Ah! Duh, yes, that would work.